Nodejs/Product/TestAdapterNetStandard/NetCoreTestDiscoverer.cs

```diff
@@ -147,10 +119,76 @@ void ProcessFiles(string fileAbsolutePath)
             }
         }
 
-        private void DiscoverTests(HashSet<string> fileList, TestFramework testFramework, ITestCaseDiscoverySink discoverySink, IMessageLogger logger, string nodeExePath, string projectFullPath)
+        private void DiscoverTests(Dictionary<string, HashSet<string>> testItems, FrameworkDiscoverer frameworkDiscoverer, ITestCaseDiscoverySink discoverySink, IMessageLogger logger, string nodeExePath, string projectFullPath)
         {
-            var discoverWorker = new TestDiscovererWorker(projectFullPath, NodejsConstants.ExecutorUri, nodeExePath);
-            discoverWorker.DiscoverTests(fileList, testFramework, logger, discoverySink);
+            foreach (var testFx in testItems.Keys)
+            {
+                var testFramework = frameworkDiscoverer.GetFramework(testFx);
+                if (testFramework == null)
+                {
+                    logger.SendMessage(TestMessageLevel.Warning, $"Ignoring unsupported test framework '{testFx}'.");
+                    continue;
+                }
+
+                var fileList = testItems[testFx];
+
+                var discoverWorker = new TestDiscovererWorker(projectFullPath, NodejsConstants.ExecutorUri, nodeExePath);
+                discoverWorker.DiscoverTests(fileList, testFramework, logger, discoverySink);
+            }
+        }
+
+        private Dictionary<string, HashSet<string>> GetConfigItems(string projectRoot)
+        {
+            var files = Directory.EnumerateFiles(projectRoot, "angular.json", SearchOption.AllDirectories)
+                .Where(x => !x.Contains("\\node_modules\\"));
+            if (files.Any())
+            {
+                return new Dictionary<string, HashSet<string>>()
+                {
+                    {
+                        TestFrameworkDirectories.AngularFrameworkName,
+                        new HashSet<string>(files)
+                    }
+                };
+            }
+
+            return default;
+        }
+
+        private Dictionary<string, HashSet<string>> GetTestItems(string projectRoot, string testFramework, string outDir)
+        {
+            // If we find ts or tsx files, get the JS file and return.
+            var files = Directory.EnumerateFiles(projectRoot, "*.ts?", SearchOption.AllDirectories)
```

Comment:
Doesn't `?` usually mean "exactly one character"?  It seems like you want "at most one character".

Reply:
Based on the Microsoft documentation, `?` means: "Zero or one character in that position."

I guess anybody can do the implementation of wildcards as they understand it. I suppose there's no standard about them.

Reply:
Weird.  Thanks for confirming.